lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -0,0 +1,379 @@
+            module.check_mode
+        )
+        self.action = module.params['action']
+        self.ca_certificates = module.params['ca_certificates']
+        self.certificate_path = module.params['certificate_path']
+        self.friendly_name = module.params['friendly_name']
+        self.iter_size = module.params['iter_size']
+        self.maciter_size = module.params['maciter_size']
+        self.passphrase = module.params['passphrase']
+        self.pkcs12 = None
+        self.privatekey_passphrase = module.params['privatekey_passphrase']
+        self.privatekey_path = module.params['privatekey_path']
+        self.src = module.params['src']
+        self.mode = module.params['mode']
+        if not self.mode:
+            self.mode = int('0400', 8)
+
+    def check(self, module, perms_required=True):
+        """Ensure the resource is in its desired state."""
+
+        state_and_perms = super(Pkcs, self).check(module, perms_required)
+
+        def _check_pkey_passphrase():
+            if self.privatekey_passphrase:
+                try:
+                    crypto_utils.load_privatekey(self.path,
+                                                 self.privatekey_passphrase)
+                    return True
+                except crypto.Error:
+                    return False
+            return True
+
+        if not state_and_perms:
+            return state_and_perms
+
+        return _check_pkey_passphrase
+
+    def dump(self):
+        """Serialize the object into a dictionary."""
+
+        result = {
+            'changed': self.changed,
+            'filename': self.path,
+        }
+        if self.privatekey_path:
+            result['privatekey_path'] = self.privatekey_path
+
+        return result
+
+    def generate(self, module):
+        """Generate PKCS#12 file archive."""
+
+        if not self.check(module, perms_required=False) or module.params['force']:
+            self.pkcs12 = crypto.PKCS12()
+
+            try:
+                self.remove()
+            except PkcsError as exc:
+                module.fail_json(msg=to_native(exc))
+
+            if self.ca_certificates:
+                ca_certs = [crypto_utils.load_certificate(ca_cert) for ca_cert
+                            in self.ca_certificates]
+                self.pkcs12.set_ca_certificates(ca_certs)
+
+            if self.certificate_path:
+                self.pkcs12.set_certificate(crypto_utils.load_certificate(
+                                            self.certificate_path))
+
+            if self.friendly_name:
+                self.pkcs12.set_friendlyname(self.friendly_name)
+
+            if self.privatekey_path:
+                self.pkcs12.set_privatekey(crypto_utils.load_privatekey(
+                                           self.privatekey_path,
+                                           self.privatekey_passphrase)
+                                           )
+
+            try:
+                pkcs12_file = os.open(self.path,
```

Comment:
Why does this not use the `with open(...) as pkcs12_file` syntax? Should be available since Python 2.5